torch/testing/_internal/common_fsdp.py

```diff
@@ -1029,7 +1029,7 @@ def _test_fsdp_parity(
         # the DDP parameters are in FP16 (from `half()`) while the FSDP
         # parameters are in FP32 (from `summon_full_params()`) and (2) DDP runs
         # the optimizer in FP16 while FSDP runs it in FP32
-        if mixed_precision is not None:
+        if mixed_precision is None:
```

Comment:
I accidentally changed the semantics of this line when refactoring a while ago. The [previous version](https://github.com/pytorch/pytorch/pull/80873/files#diff-7b5c66f99161fa6a3d9042e80f8c8cc140a64e43445feede46f55e53154f6c3dL635) used to say:
```
if not mixed_precision:
```
which is actually the opposite of
```
if mixed_precision is not None:
```